Code/Framework/AzCore/AzCore/Name/Name.h

```diff
@@ -82,62 +141,123 @@ namespace AZ
 
+        void LinkStaticName(Name** name);
+        void UnlinkStaticName();
+
+        //! If set, this name can be stored at a static scope and defer loading its NameData
+        //! until the NameDictionary is actually created. If set, this Name must hold a persistent pointer
+        //! to a string literal in m_view, and the NameDictionary will create m_data when this name is registered
+        //! and destroy m_data when it is destroyed. m_view will remain for the lifetime of this name.
+        bool m_supportsDeferredLoad = false;
+        //! If set, this name has been linked to the name list of the NameDictionary.
+        //! This means this Name can have its data stripped and restored if the NameDictionary is ever
+        //! recreated; currently, this should only occur in unit tests.
+        bool m_linkedToDictionary = false;
+
+        //! The internal hash used by this name.
+        Hash m_hash = 0;
+
         // Points to the string that represents the value of this name.
         // Most of the time this same information is available in m_data, but keeping it here too...
         // - Removes an indirection when accessing the name value.
         // - Allows functions like data() to return an empty string instead of null for empty Name objects.
         AZStd::string_view m_view;
 
-        Hash m_hash;
-
         //! Pointer to NameData in the NameDictionary. This holds both the hash and string pair.
         AZStd::intrusive_ptr<Internal::NameData> m_data;
-    };
 
+        static AZStd::thread::id s_staticNameListThread;
+        //! Describes the begin of the static list of Names that were initialized before the NameDictionary was available.
+        //! On module initialization, these names are linked into the NameDictionary's static pool and created.
+        static Name* s_staticNameBegin;
+        //! Describes the next name entry in the static list of Names.
+        Name* m_nextName = nullptr;
+        //! Describes the previous name entry in the static list of Names.
+        Name* m_previousName = nullptr;
+    };
 } // namespace AZ
 
+//! Defines a cached name literal that describes an AZ::Name. Subsequent calls to this macro will retrieve the cached name from the
+//! dictionary.
+#define AZ_NAME_LITERAL(str)                                                                                                               \
+    (                                                                                                                                      \
+        []() -> AZ::Name                                                                                                                   \
+        {                                                                                                                                  \
+            static AZ::Name nameLiteral(AZ::Name::FromStringLiteral(str));                                                                 \
```

Comment:
This should be made const as well

```suggestion
            static const AZ::Name nameLiteral(AZ::Name::FromStringLiteral(str));                                                                 \
```